qiskit/optimization/problems/name_index.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Helper Utilities """
+
+from typing import Union, List, Tuple, Dict
+
+from qiskit.optimization.utils import QiskitOptimizationError
+
+
+class NameIndex:
+    """Convert a string name into an integer index.
+    This is used for the implementation of `BaseInterface.get_indices`.
+    """
+
+    def __init__(self):
+        """Initialize a dictionary of name and index"""
+        self._dict = {}
+
+    def __contains__(self, name: str) -> bool:
+        """Check a name is registered or not.
+
+        Args:
+            name: a string name
+
+        Returns:
+            This returns True if the name has been registered. Otherwise it returns False.
+
+        """
+        return name in self._dict
+
+    def __len__(self) -> int:
+        return len(self._dict)
+
+    def build(self, names: List[str]) -> None:
+        """Build a dictionary from scratch.
+
+        Args:
+            names: a list of names
+
+        Raises:
+            QiskitOptimizationError: if any duplicate names contained in the list.
+        """
+        self._dict = {}
+        for i, name in enumerate(names):
+            if name in self._dict:
+                raise QiskitOptimizationError('Duplicate name: {}'.format(name))
+            self._dict[name] = i
+
+    def convert(self, item: Union[int, str]) -> int:
+        """Convert a name into an index.
+
+        Args:
+            item: string name or integer
+
+        Returns:
+            An index of a name
+
+        Raises:
+            QiskitOptimizationError: if argument is not included.
+        """
+
+        if isinstance(item, int):
+            if not 0 <= item < len(self._dict):
+                raise QiskitOptimizationError('Invalid index: {}'.format(item))
+            return item
+        elif isinstance(item, str):
+            if item not in self._dict:
+                raise QiskitOptimizationError('No associated index of name: {}'.format(item))
+            return self._dict[item]
+        raise QiskitOptimizationError('Invalid argument: {}'.format(item))
+
+    def convert_list(self, items: List[Union[int, str]]) -> List[int]:
```

Comment:
@Cryoris @stefan-woerner What do you prefer arguments of `convert`? To accept both an item or a list of items?

Reply:
Personally, I think `convert` is a general enough name to accept single items and list of items. Also seems more pythonic if we don't have separate methods for separate input types.

Reply:
OK